packages/cli/src/oclif/commands/scaffold.js

```diff
@@ -1,181 +1,168 @@
       );
     }
 
-    const template = await readFile(templatePath);
-    const renderTemplate = _.template(template.toString(), {
-      interpolate: /<%=([\s\S]+?)%>/g
-    });
-
-    this.startSpinner(`Writing new ${dest}.js`);
-
-    await ensureDir(path.dirname(destPath));
-    await writeFile(destPath, renderTemplate(templateContext));
+    const shouldIncludeComments = !this.flags['no-help']; // when called from other commands (namely `init`) this will be false
+    const templateContext = createTemplateContext(
+      actionType,
+      noun,
+      shouldIncludeComments
+    );
+
+    const safeNoun = templateContext.KEY;
+
+    // * create 2 new files - the scaffold and the test
+    this.log(`Adding a new ${actionType} to your project.\n`);
+    const preventOverwrite = !force;
+    // TODO: read from config file?
+
+    this.startSpinner(
+      `Creating new file: ${getLocalFilePath(newActionDir, safeNoun)}.js`
+    );
+    await writeTemplateFile(
+      actionType,
+      templateContext,
+      getFullActionFilePathWithExtension(newActionDir, safeNoun),
+      preventOverwrite
+    );
     this.stopSpinner();
-  }
-
-  async perform() {
-    const { name, type } = this.args;
 
-    if (!typeMap[type]) {
-      return this.error(
-        `Scaffold type "${type}" not found! Please see \`zaper help scaffold\`.`
-      );
-    }
-
-    try {
-      const { defaultDest, ...templateContext } = createTemplateContext(
-        type,
-        name
-      );
-      const { dest = defaultDest, entry = 'index.js' } = this.flags;
-      const entryFile = path.join(process.cwd(), entry);
+    this.startSpinner(
+      `Creating new test file: ${getLocalFilePath(
+        newTestActionDir,
+        safeNoun
+      )}.js`
+    );
+    await writeTemplateFile(
+      'test',
+      templateContext,
+      getFullActionFilePathWithExtension(newTestActionDir, safeNoun),
+      preventOverwrite
+    );
+    this.stopSpinner();
 
-      this.log(`Adding ${type} scaffold to your project.\n`);
+    // * rewire the index.js to point ot the new file
+    this.startSpinner(`Rewriting your ${entry}`);
 
-      await this.writeTemplateFile(type, templateContext, dest);
-      await this.writeTemplateFile('test', templateContext, `test/${dest}`);
+    const entryFilePath = path.join(process.cwd(), entry);
 
-      this.startSpinner(`Rewriting your ${entry}`);
+    const originalContents = await updateEntryFile(
+      entryFilePath,
+      templateContext.VARIABLE,
+      getFullActionFilePath(newActionDir, safeNoun),
+      actionType,
+      templateContext.KEY
+    );
 
-      const updatedEntry = await this.updateEntry(
-        entryFile,
-        entry,
-        templateContext.CAMEL,
-        dest
+    if (isValidAppInstall().valid) {
```

Comment:
`isValidAppInstall()` doesn't check the `skipValidInstallCheck` property, so this will run as expected. this way, we only `require` it if `node_modules` is present